types/wlr_surface.c

```diff
@@ -229,6 +236,10 @@ static void wlr_surface_move_state(struct wlr_surface *surface, struct wlr_surfa
 
 		state->buffer = next->buffer;
 		next->buffer = NULL;
+
+		wl_resource_add_destroy_listener(state->buffer, &state->buffer_destroy_listener);
```

Comment:
Causes a segfault with `weston-subsurfaces -t1 -r1` from the Weston client examples.

The buffer can be NULL here.